app/common/data/interfaces/magic_link.py

```diff
@@ -0,0 +1,37 @@
+import datetime
+import uuid
+
+from sqlalchemy import func, select, update
+
+from app.common.data.models import MagicLink, User
+from app.extensions import db
+
+
+def create_magic_link(user: User, *, redirect_to_path: str) -> MagicLink:
+    db.session.execute(update(MagicLink).where(MagicLink.user == user).values(expires_at_utc=func.current_timestamp()))
```

Comment:
Not a suggestion I've thought through at all but I wonder if these could just be deleted which would be a form of keeping the table size ~limited to the number of active users

Reply:
I quite like the idea of retaining them all for now as a bit of audit trail. I think not overly worried about table size yet, but it's good to question that thought...